src/simulations/tests/frequency_set_generator_test.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2020 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for wfa_cardinality_estimation_evaluation_framework.simulations.frequency_set_generator."""
+
+from unittest import mock
+from absl.testing import absltest
+from absl.testing import parameterized
+import numpy as np
+from wfa_cardinality_estimation_evaluation_framework.common.analysis import relative_error
+import wfa_cardinality_estimation_evaluation_framework.common.random
+from wfa_cardinality_estimation_evaluation_framework.estimators.exact_set import ExactMultiSet
+from wfa_cardinality_estimation_evaluation_framework.estimators.exact_set import LosslessEstimator
+from wfa_cardinality_estimation_evaluation_framework.simulations.frequency_set_generator import HomogeneousPmfMultiSetGenerator
+from wfa_cardinality_estimation_evaluation_framework.simulations.frequency_set_generator import HomogeneousMultiSetGenerator
+from wfa_cardinality_estimation_evaluation_framework.simulations.frequency_set_generator import HeterogeneousMultiSetGenerator
+from wfa_cardinality_estimation_evaluation_framework.simulations.frequency_set_generator import PublisherConstantFrequencySetGenerator
+
+class FrequencySetGeneratorTest(parameterized.TestCase):
+
+  def test_homogeneous_pmf_multiset_generator_single_set(self):
+      pmfgen = HomogeneousPmfMultiSetGenerator(
+          100, [2], [[1]], np.random.RandomState())
+      hists = []
+      for s in pmfgen:
+          e = ExactMultiSet()
+          e.add_ids(s)
+          hists.append(LosslessEstimator()([e]))
+      self.assertLen(hists, 1)
+      self.assertEqual(hists[0], [2])
+
+  def test_homogeneous_pmf_multiset_generator_multiple_sets(self):
+      pmfgen = HomogeneousPmfMultiSetGenerator(
+          100, [2,1,2], [[1], [0,1], [0,0,1]], np.random.RandomState())
+      sets = [s for s in pmfgen]
```

Comment:
not used?

Reply:
Thanks, you're a very sharp-eyed reviewer!